parsons/actblue/actblue.py

```diff
@@ -117,9 +127,14 @@ def poll_for_download_url(self, csv_id):
 
         logger.info("Request received. Please wait while ActBlue generates this data.")
         download_url = None
-        while download_url is None:
+        tries = 0
+        while download_url is None and (self.max_retries is None or tries < self.max_retries):
             download_url = self.get_download_url(csv_id)
             time.sleep(POLLING_DELAY)
+            tries += 1
+
+        if download_url is None:
+            raise TimeoutError("CSV generation timed out. Increase retries and try again.")
```

Comment:
While you're editing things might as well swap "increase retries" to "increase max_retries" as that's the name of the input parameter